File: digi-board/check_db.js
import pg from 'pg';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
dotenv.config({ path: path.join(__dirname, '.env') });

const pool = new pg.Pool({
    connectionString: process.env.DATABASE_URL,
});

async function check() {
    const client = await pool.connect();
    try {
        const res = await client.query(`SELECT state_classification, count(*) FROM session_telemetry WHERE session_id = 17 GROUP BY state_classification`);
        console.log("Classifications for Session 17:");
        console.table(res.rows);

        const res2 = await client.query(`SELECT focus_score, stress_score, state_classification FROM session_telemetry WHERE session_id = 17 AND focus_score < 50 LIMIT 10`);
        console.log("Samples with low focus:");
        console.table(res2.rows);
    } catch (err) {
        console.error("Failed:", err.message);
    } finally {
        client.release();
        await pool.end();
    }
}

check();

File: stress_detector/check_db.py
import pandas as pd
from sqlalchemy import create_engine
import os
from dotenv import load_dotenv

dotenv_path = os.path.join(os.path.dirname(__file__), '..', 'digi-board', '.env')
load_dotenv(dotenv_path)

DATABASE_URL = os.getenv("DATABASE_URL")
engine = create_engine(DATABASE_URL)

query = "SELECT state_classification, count(*) FROM session_telemetry WHERE session_id = 17 GROUP BY state_classification"
df = pd.read_sql(query, engine)
print("Classification counts for Session 17:")
print(df)

# Also let's check what a few rows where focus_score is low look like
query2 = "SELECT focus_score, stress_score, state_classification FROM session_telemetry WHERE session_id = 17 AND focus_score < 50 LIMIT 5"
df2 = pd.read_sql(query2, engine)
print("\nRows where focus_score < 50:")
print(df2)
